File: python/tests/model/container_test.py

from openral_py.model.container import Container


class TestContainer:

    def test_from_map_with_data(self):
        # Arrange
        uid = 123 # this is a number, not a string!
        container = Container.from_map({'UID': uid})

        # Act
        result = container.to_map()

        # Assert
        assert container.uid == uid
        assert result == {'UID': uid}

File: dart/test/model/container_test.dart
import 'package:openral_core/src/model/container.dart';
import 'package:test/test.dart';

void main() {
  group(
    Container,
    () {
      group('toMap', () {
        test('returns map', () async {
          //arrange
          final container = Container(
            uid: "uid1",
          );

          //act
          final map = container.toMap();

          //assert
          expect(
            map,
            equals({
              "UID": "uid1",
            }),
          );
        });
      });
    },
  );
}
